plugins/prometheus_plugin/metrics.hpp

```diff
@@ -311,16 +311,18 @@ struct catalog_type {
       });
 
       auto& producer = app().get_plugin<producer_plugin>();
-      producer.register_update_produced_block_metrics(
-          [&strand, this](const producer_plugin::produced_block_metrics& metrics) {
-             strand.post([metrics, this]() { update(metrics); });
-          });
       producer.register_update_speculative_block_metrics(
-              [&strand, this](const producer_plugin::speculative_block_metrics& metrics) {
+              [&strand, this](const speculative_block_metrics& metrics) {
                  strand.post([metrics, this]() { update(metrics); });
               });
-      producer.register_update_incoming_block_metrics(
-          [&strand, this](const producer_plugin::incoming_block_metrics& metrics) {
+
+      auto& chain = app().get_plugin<chain_plugin>().chain();
+      chain.register_update_produced_block_metrics(
+          [&strand, this](const produced_block_metrics& metrics) {
+             strand.post([metrics, this]() { update(metrics); });
+          });
+      chain.register_update_incoming_block_metrics(
+          [&strand, this](const incoming_block_metrics& metrics) {
```

Comment:
Could use a generic lambda instead of the 3 almost identical lambdas.

```
      auto metrics_cb =  [&strand, this](const auto& metrics) {
         strand.post([metrics, this]() { update(metrics); });
      };
      
      auto& producer = app().get_plugin<producer_plugin>();
      producer.register_update_speculative_block_metrics(metrics_cb);

      auto& chain = app().get_plugin<chain_plugin>().chain();
      chain.register_update_produced_block_metrics(metrics_cb);
      chain.register_update_incoming_block_metrics(metrics_cb);
```

Reply:
Hopefully, this will all die soon. Worth updating?

Reply:
up to you.